lib/matplotlib/cbook.py

```diff
@@ -2661,3 +2661,205 @@ def __exit__(self, exc_type, exc_value, traceback):
+      on this
+    * A callable (check_params) that returns a bool specifying if a
+      certain combination of parameters is valid.
+
+    """
+    def __init__(self, function, inverse, bounded_0_1=True, check_params=None):
+        self.function = function
+        self.inverse = inverse
+
+        if callable(bounded_0_1):
+            self._bounded_0_1 = bounded_0_1
+        else:
+            self._bounded_0_1 = lambda x: bounded_0_1
+
+        if check_params is None:
+            self._check_params = lambda x: True
+        elif callable(check_params):
+            self._check_params = check_params
+        else:
+            raise ValueError("Check params must be a callable, returning "
+                             "a boolean with the validity of the passed "
+                             "parameters, or None.")
+
+    def is_bounded_0_1(self, params=None):
+        return self._bounded_0_1(params)
+
+    def check_params(self, params=None):
+        return self._check_params(params)
+
+
+class _StringFuncParser(object):
+    """
+    A class used to convert predefined strings into
+    _FuncInfo objects, or to directly obtain _FuncInfo
+    properties.
+
+    """
+
+    _funcs = {}
+    _funcs['linear'] = _FuncInfo(lambda x: x,
+                                 lambda x: x,
+                                 True)
+    _funcs['quadratic'] = _FuncInfo(np.square,
+                                    np.sqrt,
+                                    True)
+    _funcs['cubic'] = _FuncInfo(lambda x: x**3,
+                                lambda x: x**(1. / 3),
+                                True)
+    _funcs['sqrt'] = _FuncInfo(np.sqrt,
+                               np.square,
+                               True)
+    _funcs['cbrt'] = _FuncInfo(lambda x: x**(1. / 3),
+                               lambda x: x**3,
+                               True)
+    _funcs['log10'] = _FuncInfo(np.log10,
+                                lambda x: (10**(x)),
+                                False)
+    _funcs['log'] = _FuncInfo(np.log,
+                              np.exp,
+                              False)
+    _funcs['log2'] = _FuncInfo(np.log2,
+                               lambda x: (2**x),
+                               False)
+    _funcs['x**{p}'] = _FuncInfo(lambda x, p: x**p[0],
+                                 lambda x, p: x**(1. / p[0]),
+                                 True)
+    _funcs['root{p}(x)'] = _FuncInfo(lambda x, p: x**(1. / p[0]),
+                                     lambda x, p: x**p,
+                                     True)
+    _funcs['log{p}(x)'] = _FuncInfo(lambda x, p: (np.log(x) /
+                                                  np.log(p[0])),
+                                    lambda x, p: p[0]**(x),
+                                    False,
+                                    lambda p: p[0] > 0)
+    _funcs['log10(x+{p})'] = _FuncInfo(lambda x, p: np.log10(x + p[0]),
```

Comment:
still don't think you need to bother with a log10 since you have the generic case and the user will never know (they have to write log10 anyway)